packages/embark/src/bin/embark.js

```diff
@@ -4,11 +4,18 @@
 // on babel to achieve the same goal.
 // See: https://node.green/
 
+// KEY ASSUMPTION: for a DApp to be valid, from embark's cli's perspective, it
+// must have an importable embark.config.js file or a parsable embark.json file
+// in its top-level directory; if that requirement changes in the future then
+// this script must be revised. Hypothetical example of such a change: embark
+// config info may be included in package.json under `{"embark": {...}}` -or-
+// stored in a config file.
```

Comment:
I did think even that we've discussed that one of the reasons we've introduced support of not having an `embark.json` around was to have embark work in projects that aren't embark projects per se (ie where neither an embark.json nor embark.config.js exists).

With this assumption we're neutralising that very feature but probably not a big deal at the moment as we haven't promoted it in any way.

Reply:
See: https://github.com/embarklabs/embark/pull/2260/files#diff-85b2feed7a3c1ef0d08ad0b3bfcf1ea8R413-R418

Those changes have been in place on `master` since #2260 was merged. The equivalent in this PR's refactor is:

https://github.com/embarklabs/embark/blob/e19bb39865d061f7421c1dbb2bf322902b7897a5/packages/embark/src/bin/embark.js#L404-L409

https://github.com/embarklabs/embark/blob/e19bb39865d061f7421c1dbb2bf322902b7897a5/packages/embark/src/bin/embark.js#L916-L926

My understanding of how it worked out in #2260 is that originally the plan was simply to run with the default config if no config is found in disk. That has some downsides, discussed in the PR and in team chat, so a compromise was to exit with error and prompt the user to run `embark init` (and #2260 implemented a very basic init that writes the default config into `embark.json`).

With that compromise in place, it makes the `KEY ASSUMPTION` comment still relevant w.r.t. embark's cli, so I added it back in, with some slight revisions.